servicecertsigner/v1alpha1/types.go

```diff
@@ -89,19 +73,12 @@ type ServiceCertSignerOperatorConfigSpec struct {
 	// +kubebuilder:pruning:PreserveUnknownFields
 	ServiceServingCertSignerConfig runtime.RawExtension `json:"serviceServingCertSignerConfig"`
 
-	// apiServiceCABundleInjectorConfig holds a sparse config that the user wants for this component.  It only needs to be the overrides from the defaults
-	// it will end up overlaying in the following order:
-	// 1. hardcoded default
-	// 2. this config
-	// +kubebuilder:pruning:PreserveUnknownFields
-	APIServiceCABundleInjectorConfig runtime.RawExtension `json:"apiServiceCABundleInjectorConfig"`
-
-	// configMapCABundleInjectorConfig holds a sparse config that the user wants for this component.  It only needs to be the overrides from the defaults
+	// caBundleInjectorConfig holds a sparse config that the user wants for this component.  It only needs to be the overrides from the defaults
 	// it will end up overlaying in the following order:
 	// 1. hardcoded default
 	// 2. this config
 	// +kubebuilder:pruning:PreserveUnknownFields
-	ConfigMapCABundleInjectorConfig runtime.RawExtension `json:"configMapCABundleInjectorConfig"`
+	CABundleInjectorConfig runtime.RawExtension `json:"caBundleInjectorConfig"`
```

Comment:
what does this change mean? Isn't this breaking an existing API?

Reply:
If this breaks an existing API, but the API is alpha so wouldn't that be ok?

However, maybe it makes more sense to remove this API entirely: 
 - The only thing this config is used for today is disabling authn and authz the the controller (because it doesn't expose endpoints) and setting the path of the ca bundle.
 - What is the value of supporting user overrides of these options given that they are effectively details that should be internal to the operator?
    - The lack of endpoints is an implementation detail
    - The deployment of the injector controller is entirely controlled by the operator, so overriding the bundle path from that set in the deployment would break the controller.

I'm also not clear why the bundle injecting controller shouldn't just watch the signing secret, which would remove the need for the operator to restart the controller whenever the signing secret changes. If the injection controller should not have access to the signing secret for security reasons, then the ca bundle it requires could be written to a configmap instead of being written to the signing secret. I didn't make this change in the current PR series because it seemed tangential to the goal of supporting injection for webhook configs, but I don't think it would be that complicated a change.